snuba/query/parser/validation/functions.py

```diff
@@ -18,8 +18,16 @@
 
 logger = logging.getLogger(__name__)
 
+DateTimeValidator = SignatureValidator([Column({DateTime})], enforce=False)
 
 default_validators: Mapping[str, FunctionCallValidator] = {
+    "toStartOfMinute": DateTimeValidator,
+    "toStartOfHour": DateTimeValidator,
+    "toStartOfDay": DateTimeValidator,
+    "toStartOfYear": DateTimeValidator,
+    # Techincally this can be called with either a datetime or string
+    # but seems okay to be more restrictive for now
+    "toUnixTimestamp": DateTimeValidator,
```

Comment:
Only place in Sentry i could find this being used so I think restricting it to just the datetime param should be ok https://github.com/getsentry/sentry/blob/f5ecc4ae0ac9e760ac52ddee88acd0e1df66e301/src/sentry/search/events/fields.py#L2172